### PR TITLE
support openjdk for rpm.

### PR DESCRIPTION
**Pull request by toyama0919**
Hello.

I use openjdk.
When it is openjdk, cannot install it in RPM.
## openjdk

```
$ java -version
openjdk version "1.8.0_91"
OpenJDK Runtime Environment (build 1.8.0_91-b14)
OpenJDK 64-Bit Server VM (build 25.91-b14, mixed mode)
```
## oracle

```
$ java -version
java version "1.8.0_77"
Java(TM) SE Runtime Environment (build 1.8.0_77-b03)
Java HotSpot(TM) 64-Bit Server VM (build 25.77-b03, mixed mode)
```

### CODE REVIEW

**Comment by martint** [comment]
@cawallin, can you review this?

**Comment by billonahill** [comment]
@toyama0919 just curious, what do you get when you run this command with open JDK:

`java -XshowSettings:properties -version 2>&1 | grep vendor`

I'd like to see if the proposed change to JVM vendor checking discussed [here](https://groups.google.com/forum/#!topic/presto-users/U4QgYcpp6wM) would work. See https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/server/PrestoSystemRequirements.java#L43.

**Comment by cawallin** [comment]
@toyama0919 looks good!

@billonahill:
I went ahead and ran that, and got:
    /usr/lib/jvm/java-8-openjdk-amd64/bin/java -XshowSettings:properties -version 2>&1 | grep vendor
    java.specification.vendor = Oracle Corporation
    java.vendor = Oracle Corporation
    java.vendor.url = http://java.oracle.com/
    java.vendor.url.bug = http://bugreport.sun.com/bugreport/
    java.vm.specification.vendor = Oracle Corporation
    java.vm.vendor = Oracle Corporation

But I think it would make sense to merge this commit first, and then (if everyone agrees) you can add the vendor check both to PrestoServerRequirements and this Java check.

**Comment by toyama0919** [comment]
@cawallin @billonahill
I felt that the check Oracle was unnecessary.
I want remove...(if everyone agrees)

**Comment by billonahill** [comment]
+1 for merging this as is. I didn't mean to hijack the PR, but wanted to hear thoughts on the other issue. :)

I'm fine with removing the Oracle check too, or changing it to assert against jvm.vendor.specification.

**Comment by toyama0919** [comment]
@billonahill I see😊

**Comment by cawallin** [comment]
@toyama0919 -- the reason that we check for Oracle as a vendor is because Presto relies on the innards of the HotSpot JVM, so it's unclear whether Presto would function with Java that is not Oracle or OpenJDK. Or at least that's what I understand from what @martint and @dain have said, if necessary they can clarify.

**Comment by dain** [comment]
The biggest reason we have these checks is to keep support requests down.  For example, Redhat used to ship with a super old and broken JVM.  Additionally, Presto makes use of some internal APIs in Oracle's JVM.  These APIs are in OpenJDK but I'm not sure they are in IBMs JDK.  

Instead of removing the check, I would be more comfortable expanding the whitelist to OpenJDK.  Also, we should have some process to qualifying a JVM implementation for Presto.

**Comment by billonahill** [comment]
@dain what if we changed the check to go against `java.specification.vendor` instead of `java.vendor`? It looks like [IBM](https://www.ibm.com/support/knowledgecenter/ssw_i5_54/rzaha/sysprop2.htm) has that set to "Sun Microsystems, Inc." so the check would correctly fail. This would allow extending the JVM with additional functionality, while still being to the Oracle spec, which we do:
```
    java.specification.vendor = Oracle Corporation
    java.vendor = Twitter
```

If folks are ok with this I can submit a separate PR.

**Comment by toyama0919** [comment]
@dain 
>The biggest reason we have these checks is to keep support requests down. For example, Redhat used to ship with a super old and broken JVM. Additionally, Presto makes use of some internal APIs in Oracle's JVM. These APIs are in OpenJDK but I'm not sure they are in IBMs JDK.


I understood it.

**Comment by toyama0919** [comment]
@martint ping

**Comment by martint** [comment]
I think it's reasonable to allow OpenJDK. It is, for all intents and purposes, the same code as Oracle JDK. We've also used it in the past (and will likely use it in the future) to validate releases of the JDK before there was an official build.

**Comment by billonahill** [comment]
@martint what are your thoughts on validating against `java.specification.vendor` instead of `java.vendor`. That would cover OpenJDK, Oracle JDK and anyone else who extends/implements the Oracle spec (see my comment above).

**Comment by martint** [comment]
@billonahill, that one is a bit more debatable. We only test Presto against OpenJDK/Oracle implementation, which is what the "java.vendor" captures. Presto relies on internal OpenJDK APIs and is tuned for certain implementation details (how Hotspot optimizes certain code, GC quirks, etc). So, technically, it's not enough to guarantee that an implementation complies with the spec.

One alternative could be to add an option for presto to skip requirements checks -- a kind of "I know what I'm doing, don't bug me" mode.

**Comment by billonahill** [comment]
Thanks @martint makes sense. We can pursue an "I know what I'm doing" patch as a separate PR.

**Comment by toyama0919** [comment]
@martint 

```
  * Presto supports only Oracle Java
```

Is this OK?
it is meaning 「Oracle java」 include open-jdk.

**Comment by dabaitu** [comment]
@martint - I'm trying to implement the skip jvm check config setting that you and @billonahill have discussed.
Not sure if it's appropriate to ask for some help here, but I figured I'd try since this thread has the full context. I also asked in presto users group.

There's not much doc out there on the airlift framework of configuration. All examples I find and also the presto code have only the modules read the configs, and from airlift source code I haven't found a natural way to retrieve some config settings before all modules get initialized. Something related to system requirements should probably be checked before the modules initialize. I was also looking for a function in Bootstrap in airlift to see if module initialization can be ordered, so I can add a module to check system requirements, but didn't find anything like that.
My hunch tell me I'm not finding something that's already there in airlift; if anyone could give me some tips that would be great.

**Comment by martint** [comment]
Sorry, I didn't see the notification for your comment.

While you can use Airlift's configuration system in standalone mode (by creating a ConfigurationFactory), you'd need to know how to obtain the map of properties. Presto relies on some of the conventions established by other Airlift components (launcher, bootstrap, packaging, etc). It'd be tricky to do that outside of Bootstrap because of the way it interacts with VM-wide APIs, such as logging, jmx, etc.

Unfortunately, there's no good solution to this without significant changes to Airlift. There's also a bit of chicken-and-egg issue: you want to keep the code that performs the checks as simple as possible to make them as compatible as possible with the largest number of implementations. One could argue that the checks should be done in code that's compiled against an early version of the JDK to avoid failing with cryptic "class format version" errors.

I can think of two alternatives:
- Explicitly read a system property. This means that the way to configure it would be by adding -Dxxx=yyy to jvm.config when deploying a server.
- Bite the bullet and allow any JVMs other than those we've identified as being problematic. Given that this set is (or should be) quite small, it would seem like a reasonable compromise.

**Comment by dabaitu** [comment]
thanks @martint for the suggestion. Will look into it!

File: README.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ See the [User Manual](https://prestodb.io/docs/current/) for deployment instruct
 
 * Mac OS X or Linux
 * Java 8 Update 60 or higher (8u60+), 64-bit
+  * Presto supports only Oracle Java
```
Comment (cawallin) [inline README.md]: Presto supports Oracle Java and OpenJDK

**Comment by anusudarsan** [comment]
@martint Can we merge this after rebasing README?

**Comment by anusudarsan** [comment]
rebased and reworded commit message here - https://github.com/prestodb/presto/pull/10328

**Pull request closed without merging**